test/ngMessageFormat/messageFormatSpec.js

```diff
@@ -0,0 +1,699 @@
+'use strict';
+
+/* TODO: Add tests for:
+ • Whitespace preservation in messages.
+ • Whitespace ignored around syntax except for offset:N.
+ • Escaping for curlies and the # symbol.
+ • # symbol value.
+ • # symbol value when gender is nested inside plural.
```

Comment:
Poring over what little information is there on the ICU site does not shed information on whether this should be valid.  The 3rd party JS lib supports "#" even in select without a plural.  That might be a bug.  I don't like the "#" in general because it has a bunch of limitations.  So this won't work today and if it break someone, they can file a bug and I'll fix it.

Looking at the actual ICU MessageFormat source test file, `test/intltest/tmsgfmt.cpp`, I don't see any `select` message using the `#` symbol, only `plural`s and `selectordinal` (which we don't support.)  e.g. line 1863

```cpp
void TestMessageFormat::TestSelectOrdinal() {
    IcuTestErrorCode errorCode(*this, "TestSelectOrdinal");
    // Test plural & ordinal together,
    // to make sure that we get the correct cached PluralSelector for each.
    MessageFormat m(
        "{0,plural,one{1 file}other{# files}}, "
        "{0,selectordinal,one{#st file}two{#nd file}few{#rd file}other{#th file}}",
        Locale::getEnglish(), errorCode);
    if (errorCode.logDataIfFailureAndReset("Unable to instantiate MessageFormat")) {
        return;
    }
```

Let's leave this "unspecified" for now until I can dig into the gory details of the actual implementation to see if it's "really really" supposed to work in select messages.